plutus-contract/src/Plutus/Contract/StateMachine.hs

```diff
@@ -241,6 +242,17 @@ waitForUpdateUntil StateMachineClient{scInstance, scChooser} timeoutSlot = do
                 Left err         -> throwing _SMContractError err
                 Right (state, _) -> pure $ WaitingResult (tyTxOutData state)
 
+-- | Same as 'waitForUpdateUntilSlot', but works with 'POSIXTime' instead.
+waitForUpdateUntilTime ::
+    ( AsSMContractError e
+    , AsContractError e
+    , PlutusTx.IsData state
+    )
+    => StateMachineClient state i
+    -> POSIXTime
+    -> Contract w schema e (WaitingResult state)
+waitForUpdateUntilTime sm timeoutTime = do
+    waitForUpdateUntilSlot sm $ TimeSlot.posixTimeToSlot timeoutTime
```

Comment:
Does this work reliably? I guess it'll maybe just need to be tweaked when we make the slot length configurable. But generally `posixTimeToSlot` isn't well-defined: you can do `firstSlotAfterTime` or `lastSlotBeforeTime` or something, but not every time corresponds to a slot!

Reply:
In my understanding, if 1 slot has length of 2, then `Time 0 == Slot 0`, `Time 1 == Slot 0` and `Time 2 == Slot 1`, since a slot is a range of time.

Reply:
But yeah, we'll have to address this in the next PR when slot length becomes configurable.

Reply:
That's not what `posixTimeToSlot` does, though? It just subtracts the time from the start time and calls it a slot!

Reply:
True for now. The PR #3351 will change this.